Ct 1876 non trigger extend for pit

beccapearce opened the pull request.

misaka commented on app/state_machines/cases/foi_state_machine.rb:
```diff
@@ -487,8 +487,16 @@ def ready_to_send?(kase)
-  def message_sent_by_responder?(user, kase)
-    user == kase.responder_assignment.user
+  def able_to_send?(user, kase)
+    message_not_sent_by_responder?(user, kase) && case_has_responder(kase)
```
So `case_has_responder` seems to be protecting to ensure that the case actually has a responder assigned to it, but there could be the situation where the responding team assigned to a case has rejected the case, and no new one has been assigned. I think that that situation would case an error in the code below, since `kase.responding_assignment` would return a `nil` so `kase.responding_assignment.user` would break. Can you confirm whether this is an issue, and if so protect against it?

misaka replied:
Here's the steps I envision that could cause a problem:

1. Case is created and assigned to a BU
2. BU rejects the case. The `responding_assignment` for the case is then set to `rejected`.
3. Disclosure BMT adds a message to the case before re-assigning it, this code then gets `No method 'user' for nil` kind-of error.

Commit:
CT-1876 allow disclosure to extend for PIT on standard cases

Commit:
CT-1876 allow managers to extend for pit instead of specialist

An issue arose where non_trigger FOIs could not be
extended_for_pit since they needed an assigned_disclosure_specialist
and as they have no disclosure approvers the action was not permitted.
To solve this problem we have changed the extend for pit action to be
permitted to any manager on a case.

stephenrichards reviewed (changes requested):
Code OK.
Although I can see what has been done, I think we should have a summary of what the issue was and what we did to resolve it on the commit message and/or ticket itself.
